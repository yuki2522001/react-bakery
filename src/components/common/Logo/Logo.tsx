import React, { memo } from "react";
import { Link } from "react-router-dom";
import "./logo.css";

interface LogoProps {
  src: string;
  alt?: string;
  productId?: string;
}

const Logo: React.FC<LogoProps> = memo(({ src, alt }) => {
  return (
    <Link data-testid="logo" to="/">
      <img className="logo" src={src} alt={alt} />
    </Link>
  );
});

export default Logo;
