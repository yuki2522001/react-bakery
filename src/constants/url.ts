const PRODUCTS_URL = `${process.env.VITE_BASE_URL}/products`;
const CATEGORIES_URL = `${process.env.VITE_BASE_URL}/categories`;

export { PRODUCTS_URL, CATEGORIES_URL };