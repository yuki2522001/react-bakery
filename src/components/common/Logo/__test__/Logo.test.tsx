import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import url from "../../../../assets/images/logos/logos.png";
import Logo from "../Logo";

describe("Logo component", () => {
  test("should render logo component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Logo src={url} />
      </Router>
    );
    expect(getByTestId("logo")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Logo src={url} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
