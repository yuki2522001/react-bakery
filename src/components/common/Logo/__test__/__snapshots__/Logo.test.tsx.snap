// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component matches snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="logo"
    href="/"
  >
    <img
      class="logo"
    />
  </a>
</DocumentFragment>
`;
